lib/mirage.ml

```diff
@@ -471,10 +472,14 @@ module Entropy = struct
     append_main "module %s = %s" (module_name t) (construction t) ;
     newline_main () ;
     append_main "let %s () =" (name t);
-    append_main "  %s.connect %s" (module_name t) (id t);
+    append_main "  %s.connect %s >>= function" (module_name t) (id t);
+    append_main "  | `Error e -> %s" (driver_initialisation_error "entropy");
+    append_main "  | `Ok entropy ->";
+    append_main "  %s.handler entropy Nocrypto.Rng.Accumulator.add_rr >>= fun () ->" (module_name t);
```

Comment:
so `~source` is added by mirage-entropy ([unix](https://github.com/mirage/mirage-entropy/blob/master/unix/entropy_unix.ml#L66) and [xen](https://github.com/mirage/mirage-entropy/blob/master/xen/entropy_xen.ml#L165) version).